src/ai/ai_help_structs.cc

```diff
@@ -382,7 +383,7 @@ MineableField::MineableField(const Widelands::FCoords& fc)
 }
 
 EconomyObserver::EconomyObserver(Widelands::Economy& e)
-   : economy(e), dismantle_grace_time(Time()), fields_block_last_time(Time(0)) {
+   : economy(e), fields_block_last_time(Time(0)) {
```

Comment:
What (which check) is the reasonm for deleting this initialization with indefinite time (highest value possible).   
I am not sure if this variable is proper initialized without this.

Reply:
When a member variable is not initialized explicitly but has a default constructor then it will be initialized by implicitly calling the default constructor. What we do here though is to instantiate it with the copy constructor that copies a default-constructed temporary variable. So by removing the explicit constructor call the default constructor will be used.

Reply:
the problem is I couldn't find a proper initializing default constructor for this. Furthermore I don't understand why only this single parameter is omitted and the others are not.

Reply:
There it is:
https://github.com/widelands/widelands/blob/eda47cde863b64577473b181950f5900bc4dee92/src/base/times.h#L145
This ctor has a default value as its only argument and is therefore recognized as the default constructor.

> Furthermore I don't understand why only this single parameter is omitted and the others are not.

I'm not sure what you mean exactly? 
When we default-initialize a member variable (passing zero arguments to the constructor), then and only then we can omit the call altogether and the default constructor will be called implicitly.

Reply:
ok got it finally. Thanks for the patience